src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,35 +58,44 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${id}`}
             </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
+              aria-label="modal-close"
               type="button"
               className="delete"
               data-cy="modal-close"
+              onClick={() => onClose()}
             />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {title}
             </p>
 
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
+            <p
+              className="block"
+              data-cy="modal-user"
+            >
+              <strong
+                className={classNames(
+                  { 'has-text-success': completed },
+                  { 'has-text-danger': !completed },
+                )}
+              >
+                {completed ? 'Done' : 'Planned'}
+              </strong>
               {' by '}
 
               <a href="mailto:Sincere@april.biz">
-                Leanne Graham
+                {user?.name}
               </a>
```

Comment:
when there is no name, then there is no sense to render an empty link. Try to avoid redundant html generation. And you missed to update the href attribute value, we don't want one user to get all emails.
```
 {user?.name && 
     <> 
         by 
         <a href={`mailto:${user.email}`}>
            {user.name}
         </a>
     </>
 }
```